### ISSUE TITLE
Setup production grade webserver for hosting on live node

### ISSUE THREAD

**Issue by cshinaver** [ccastil1/SoftwareEngineering#2]
Currently uses development one. Should also setup a systemd init script so it starts on boot. Maybe also something that monitors it later down the line

**Comment by cshinaver** [comment]
Progress: 
Goal: fully deploy to live server with git push live 
Looking into Puma and Capistrano. Lots of stuff in setup that isn't needed.
Will probably roll own deployment strategy using git hooks. Got basic hook working that updates remote repo when changes pushed to it. Now just going to attempt to get a good webserver (nginx?) running the rails box. 

Annoying thing is repetability. Best case is get everything into chef scripts. We'll see...

**Comment by cshinaver** [comment]
Ugh. Tutorials make way too many assumptions. Anyway, I think this is the strategy:

* Web server: nginx
* App server: Puma
* Deployment administrator: Capistrano

Pretty standard Rails setup. I think putting in the effort to use Capistrano now will pay off when moving to multiple nodes.

**Comment by cshinaver** [comment]
TODO: add nginx to chef scripts for live node

**Comment by cshinaver** [comment]
Had to get rid of subdirectory se_filesystem and move the files up for the deploy to work. Other solutions were disgusting. Convention over configuration I guess.

**Comment by cshinaver** [comment]
Well, this has been terrible. 

More or less followed [this](https://www.digitalocean.com/community/tutorials/deploying-a-rails-app-on-ubuntu-14-04-with-capistrano-nginx-and-puma) tutorial. I deviated a lot however, and it was not a great time. For example, I did not use rvm. Just system ruby. I also ignored all the ssh keys stuff and database stuff etc etc. Putting it here for reference.

For some reason, the `cap production puma:start` command does not work. I currently have to run this disgusting thing
`  bundle exec puma -b 'unix:///home/ubuntu/apps/se_filesystem/shared/tmp/sockets/puma.sock'  -e production  --control 'unix:///home/ubuntu/apps/se_filesystem/shared/tmp/sockets/pumactl.sock' -S /home/ubuntu/apps/se_filesystem/shared/tmp/pids/puma.state `
from within the ~/apps/se_filesystem/current directory (within a screen session so it persists) before trying to deploy, or the system will not be able to restart the app server properly. 

Deploy works however. After code is pushed to master (on Github), running `cap production deploy` with pull down the latest version of master from github to the server, create a release, recompile assets, run migrations, etc etc and restart the app server. One command (ish) deploy. Yay. 

Improvements will be to get the puma server to start on boot with a systemd init script.
